### PR TITLE
CUE loading unit tests

### PR DESCRIPTION
**Pull request by danielpcox**
[sc-22033]

### CODE REVIEW

File: pkg/cuemodule/load_test.go [hunk 1]
```diff
@@ -1,40 +1,157 @@
...
 	}
+	ctx = context.Background()
+}
 
-	t.Run("TestExtractConfig", func(t *testing.T) {
-		config, _ := operatorCUE.ExtractConfig()
+func TestLoad(t *testing.T) {
+	config, defaults := operatorCUE.ExtractConfig() // FUT
 
+	t.Run("TestExtractedConfigAndDefaults", func(t *testing.T) {
```
Comment (alecholmez) [inline pkg/cuemodule/load_test.go:45]: It looks like you're executing quite a few tests inside of TestLoad. Could these be converted to table driven tests?

https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Comment (danielpcox) [reply pkg/cuemodule/load_test.go:45]: How about we do that later so we can get these tests in? I don't offhand remember right now if we could have used table-driven tests or not for this, but I don't think we're actually _changing_ any inputs in this set of tests, just confirming several different things from the same input.

File: pkg/cuemodule/load.go [hunk 2]
```diff
@@ -271,13 +272,15 @@ type justKeys struct {
 	CatalogZoneKey   string `json:"zone_key"`   // CatalogService
 }
 
+var unrecognizedKindLabel = "unrecognized"
```
Comment (alecholmez) [inline pkg/cuemodule/load.go:275]: Can this be a `const`?

File: pkg/cuemodule/load_test.go (hunk 1 above)
Comment (alecholmez) [reply pkg/cuemodule/load_test.go:45]: Okay no worries. I just read through it and I think this is okay.

File: pkg/cuemodule/load_test.go [hunk 3]
```diff
@@ -1,40 +1,160 @@
...
+		defaults.SidecarList = []string{"foo", "bar"}
+		tempOperatorCUE, err := operatorCUE.TempGMValueUnifiedWithDefaults(defaults) // FUT 1
+		if err != nil {
+			t.Fatal(err)
+		}
+		redisListener, err := tempOperatorCUE.ExtractRedisListener() // FUT 2
+		if err != nil {
+			t.Fatal(err)
+		}
+		//t.Fatal(string(redisListener))
```
Comment (alecholmez) [inline pkg/cuemodule/load_test.go:68]: Delete this?

File: pkg/cuemodule/load_test.go [hunk 4]
```diff
@@ -1,40 +1,160 @@
...
+		assert.Equal(t, expectedSpireBlock, defaults.Spire)
+	})
+
+	t.Run("Test defaults injection for sidecar list", func(t *testing.T) {
+		defaults.SidecarList = []string{"foo", "bar"}
+		tempOperatorCUE, err := operatorCUE.TempGMValueUnifiedWithDefaults(defaults) // FUT 1
+		if err != nil {
+			t.Fatal(err)
+		}
+		redisListener, err := tempOperatorCUE.ExtractRedisListener() // FUT 2
```
Comment (alecholmez) [inline pkg/cuemodule/load_test.go:64]: Not sure what this means:
```
// FUT 2
```

File: pkg/cuemodule/load_test.go [hunk 5]
```diff
@@ -1,40 +1,160 @@
...
+			Namespace:    "spire",
+			TrustDomain:  "greymatter.io",
+		}
+		assert.Equal(t, expectedSpireBlock, defaults.Spire)
+	})
+
+	t.Run("Test defaults injection for sidecar list", func(t *testing.T) {
+		defaults.SidecarList = []string{"foo", "bar"}
+		tempOperatorCUE, err := operatorCUE.TempGMValueUnifiedWithDefaults(defaults) // FUT 1
+		if err != nil {
```
Comment (alecholmez) [inline pkg/cuemodule/load_test.go:61]: Wherever you're checking for `if err != nil` lets use assert instead:
```go
assert.NoError(t, err)
```

File: pkg/cuemodule/load_test.go [hunk 6]
```diff
@@ -1,40 +1,160 @@
...
+		}
+		redisListener, err := tempOperatorCUE.ExtractRedisListener() // FUT 2
+		if err != nil {
+			t.Fatal(err)
+		}
+		//t.Fatal(string(redisListener))
+		res := gjson.GetBytes(redisListener, "secret.subject_names")
+		subjects := res.Array()
+		for i, clusterName := range defaults.SidecarList {
+			expectedSubject := helperSpireSubject(
```
Comment (alecholmez) [inline pkg/cuemodule/load_test.go:72]: It's generally idiomatic go to do this format instead:
```go
			assert.Equal(
				t,
				helperSpireSubject(
				    defaults.Spire.TrustDomain,
				    config.OperatorNamespace,
				    mesh.Name,
				    clusterName,
                ),
				subjects[i].String(),
            )
```

Comment (alecholmez) [reply pkg/cuemodule/load_test.go:72]: If you'd like to keep the separate variable that also works too but the real nitpick I have is this `)` on the final line of the function call needs its own separate line

File: pkg/cuemodule/load_test.go [hunk 7]
```diff
@@ -1,40 +1,160 @@
...
+			if env.Name == "XDS_CLUSTER" {
+				xdsClusterExists = true
+				assert.Equal(t, env.Value, clusterLabel, "XDS_CLUSTER label does not match clusterLabel")
+			}
 		}
-		tempOperatorCUE, _ := operatorCUE.TempGMValueUnifiedWithDefaults(defaults)
-		redisListener, _ := tempOperatorCUE.ExtractRedisListener()
-		logger.Info("Loaded Redis listener", "listener", redisListener)
-		//logger.Info("LoadAll sidecarList", "SidecarList", defaults.SidecarList)
+		if !xdsClusterExists {
```
Comment (alecholmez) [inline pkg/cuemodule/load_test.go:115]: This can also be an assert

File: pkg/cuemodule/load_test.go [hunk 8]
```diff
@@ -1,40 +1,160 @@
...
+				//fmt.Println(res.String())
+				assert.Contains(t, resCluster.String(), sidecarName)
+			}
+			resPort := gjson.GetBytes(o, "instances.0.port")
+			if resPort.Int() != 0 {
+				sidecarPortExists = true
+				assert.Equal(t, int64(sidecarPort), resPort.Int())
+			}
+		}
+		if !sidecarPortExists {
```
Comment (alecholmez) [inline pkg/cuemodule/load_test.go:146]: Assert?

**Review by alecholmez — CHANGES_REQUESTED** [review]
Just some minor things but overall tests look great

File: pkg/cuemodule/load_test.go (hunk 4 above)
Comment (danielpcox) [reply pkg/cuemodule/load_test.go:64]: Function Under Test

**Review by alecholmez — APPROVED** [review]
Looks good!

**Comment by danielpcox** [comment]
Oh hey, nice job @joel-epstein! Please do the honors.

**Pull request merged**